frontends/web/src/routes/device/waiting.tsx

```diff
@@ -14,71 +14,62 @@
+import { getTesting } from '../../api/testing';
 import { Entry } from '../../components/guide/entry';
 import { Guide, store as panelStore } from '../../components/guide/guide';
 import { AppLogo, SwissMadeOpenSource } from '../../components/icon/logo';
 import { Footer, Header } from '../../components/layout';
 import { setSidebarStatus } from '../../components/sidebar/sidebar';
-import { load } from '../../decorators/load';
-import { translate, TranslateProps } from '../../decorators/translate';
 import { debug } from '../../utils/env';
-import style from './bitbox01/bitbox01.module.css';
 import { SkipForTesting } from './components/skipfortesting';
+import style from './bitbox01/bitbox01.module.css';
 
-interface TestingProps {
-    testing?: boolean;
-}
-
-type WaitingProps = TestingProps & TranslateProps;
+export const Waiting: FunctionComponent = () => {
+    const { t } = useTranslation();
+    const testing = useLoad(getTesting);
 
-class Waiting extends Component<WaitingProps> {
-    public UNSAFE_componentWillMount() {
+    useEffect(() => {
         const { sidebarStatus } = panelStore.state;
         if (['forceCollapsed', 'forceHidden'].includes(sidebarStatus)) {
             setSidebarStatus('');
         }
-    }
+    }, []);
 
-    public render() {
-        const { t, testing } = this.props;
-        return (
-            <div className="contentWithGuide">
-                <div className="container">
-                    <Header title={<h2>{t('welcome.title')}</h2>} />
-                    <div className="content padded narrow isVerticallyCentered">
-                        <div>
-                            <AppLogo />
-                            <div className="box large">
-                                <h3 className={style.waitingText}>{t('welcome.insertDevice')}</h3>
-                                <p className={style.waitingDescription}>{t('welcome.insertBitBox02')}</p>
-                            </div>
-                            {
-                                testing && (
-                                    <div className={style.testingContainer}>
-                                        <SkipForTesting show={!!testing} />
-                                    </div>
-                                )
-                            }
+    return (
+        <div className="contentWithGuide">
+            <div className="container">
+                <Header title={<h2>{t('welcome.title')}</h2>} />
+                <div className="content padded narrow isVerticallyCentered">
+                    <div>
+                        <AppLogo />
+                        <div className="box large">
+                            <h3 className={style.waitingText}>{t('welcome.insertDevice')}</h3>
+                            <p className={style.waitingDescription}>{t('welcome.insertBitBox02')}</p>
                         </div>
+                        {
+                            (debug && testing) && (
```

Comment:
Minor difference, before the HOC querying `/testing` conditionally, only if debug was true.

Now it is always querying `/testing` and the condition is in the rendering.

Reply:
Why not

`const testing = debug && useLoad(getTesting);` above, so the call only happens in debug? would that work? (useEffect etc are still a bit mysterious to me)

edit: i tried and this happend

> React Hook "useLoad" is called conditionally. React Hooks must be called in the exact same order in every component rende

This would work instead but is ugly:

    const testing = useLoad(debug ? getTesting : () => Promise.resolve(false));

Reply:
yes it's a bit ugly 🤔 

ugly alternative

```js
const resolvePromise = (value: any) => Promise.resolve(value);

const testing = useLoad(debug ? getTesting : resolvePromise(false));

```